src/main/java/org/jenkinsci/plugins/lucene/search/Field.java

```diff
@@ -18,69 +21,73 @@
 public enum Field {
     ID("id", DefaultSearchable.FALSE, Numeric.TRUE, Persist.TRUE) {
         @Override
-        public String getValue(AbstractBuild<?, ?> build) {
+        public String getValue(Run<?, ?> build) {
             return build.getId();
         }
     },
     PROJECT_NAME("projectname", Persist.TRUE) {
-        public String getValue(final AbstractBuild<?, ?> build) {
+        public String getValue(final Run<?, ?> build) {
             StringBuilder builder = new StringBuilder();
-            if (!build.getProject().getParent().getDisplayName().equalsIgnoreCase("jenkins")) {
-                builder.append(build.getProject().getParent().getFullName() + "/");
+            if (!build.getParent().getParent().getDisplayName().equalsIgnoreCase("jenkins")) {
+                builder.append(build.getParent().getParent().getFullName() + "/");
             }
-            builder.append(build.getProject().getName());
+            builder.append(build.getParent().getName());
             return builder.toString();
         }
     },
 
     PROJECT_DISPLAY_NAME("projectdisplayname", Persist.TRUE) {
         @Override
-        public String getValue(AbstractBuild<?, ?> build) {
+        public String getValue(Run<?, ?> build) {
             StringBuilder builder = new StringBuilder();
-            if (!build.getProject().getParent().getDisplayName().equalsIgnoreCase("jenkins")) {
-                builder.append(build.getProject().getParent().getDisplayName() + "/");
+            if (!build.getParent().getParent().getDisplayName().equalsIgnoreCase("jenkins")) {
+                builder.append(build.getParent().getParent().getDisplayName() + "/");
```

Comment:
Why not <code>build,getParent().getFiullDisplayName()</code> ? The logic here does not seem to be compatible with Folders Plugin